app/colorpalettewidget.cpp

```diff
@@ -294,6 +294,66 @@ void ColorPaletteWidget::updateGridUI()
+
+    // Separate rgb values for convenience
+    const int r = c.red(),
+                g = c.green(),
+                b = c.blue();
+
+    // Convert RGB to XYZ with D65 white point
+    const qreal x = 0.412453*r + 0.357580*g + 0.180423*b,
+                y = 0.212671*r + 0.715160*g + 0.072169*b,
+                z = 0.019334*r + 0.119193*g + 0.950227*b;
+
+    // Convert XYZ to CEI L*u*v
+    // Helper function for the conversion
+    auto f = [] (const double a) { return a > 0.008856 ? cbrt(a) : 7.787*a + 16/116; };
+    // XYZ tristimulus values for D65
+    const qreal xn = 95.047,
+                yn = 100,
+                zn = 108.883;
+    const qreal l = y/yn > 0.008856 ? 116*cbrt( y/yn ) - 16 : 903.3*y/yn,
+                u = 500 * ( f( x/xn ) - f( y/yn ) ),
+                v = 200 * ( f( y/yn ) - f( z/zn ) );
```

Comment:
It’s been some time since I last worked with colorspace math so I won’t even begin to pretend that I understand this, but either way there are way too many magic numbers in here for my liking. It’d be great if you could state the reference/source at the very least.